tensorflow/contrib/learn/python/learn/learn_io/data_feeder.py

```diff
@@ -507,10 +507,13 @@ def _feed_dict_fn():
           inp[i, :] = six.next(self._x)
         except StopIteration:
           self.stopped = True
-          inp = inp[:i, :]
-          if self._y is not None:
-            out = out[:i]
-          break
+          if i==0:
+            raise StopIteration
```

Comment:
@martinwicke - Not sure if I understand your question very well. I presume StopIteration is automatically picked by `super_session.should_stop()` in graph_action.py line 287